File: src/main/java/com/koddy/server/member/application/usecase/query/response/MentoringPeriodResponse.java
package com.koddy.server.member.application.usecase.query.response;

import com.koddy.server.member.domain.model.mentor.MentoringPeriod;

import java.time.LocalDate;

public record MentoringPeriodResponse(
        LocalDate startDate,
        LocalDate endDate
) {
    public static MentoringPeriodResponse of(final MentoringPeriod mentoringPeriod) {
        if (mentoringPeriod == null) {
            return null;
        }
        return new MentoringPeriodResponse(mentoringPeriod.getStartDate(), mentoringPeriod.getEndDate());
    }
}

File: src/main/kotlin/com/koddy/server/member/domain/model/response/MentoringPeriodResponse.kt
package com.koddy.server.member.domain.model.response

import com.koddy.server.member.domain.model.mentor.MentoringPeriod
import java.time.LocalDate

data class MentoringPeriodResponse(
    val startDate: LocalDate,
    val endDate: LocalDate,
) {
    companion object {
        fun from(mentoringPeriod: MentoringPeriod?): MentoringPeriodResponse? {
            if (mentoringPeriod == null) {
                return null
            }

            return MentoringPeriodResponse(
                startDate = mentoringPeriod.startDate,
                endDate = mentoringPeriod.endDate,
            )
        }
    }
}
